samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManagerRestoreUtil.java

```diff
@@ -185,17 +185,6 @@ private static CompletableFuture<Map<TaskName, Checkpoint>> restoreAllTaskInstan
               throw new SamzaException(msg, ex); // wrap in unchecked exception to throw from lambda
             }
           }
-
```

Comment:
Could we chain the delete future to L:156  taskrestoremanager.restore() instead?

Reply:
@dxichen thanks for the review. Sorry I am unsure which future are you referring to. If you meant close() -> that can't be chained because in case of exception, if we retry the restore, the restore manager would be prematurely closed. So we only close it if we are sure the future is complete.
If you meant something else, please let me know and I can address/clarify it.

Reply:
by future here I mean if we chain `restoreDeletedSnapshots` future to the restore such that it is always performed after restore and before the close call avoiding the race condition during restore

Reply:
Thanks. Discussed this with Prateek and updated the logic to make it clearer. We might have previously missed a bug. Please let me know if it looks okay now.